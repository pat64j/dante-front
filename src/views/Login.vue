<template>
  <v-container fluid class="fill-height pa-0">
    <v-row no-gutters class="fill-height">
      <v-col class="d-none d-md-block" md="5">
        <div class="bg-img-wrapper fill-height">
          <v-row no-gutters>
            <v-col col="12">
              <div class="top-logo-display">
                <img
                  style="margin: 0 !important"
                  :src="require('../assets/svg/church-logo.svg')"
                  class="my-3"
                  contain
                  height="150"
                  width="150"
                />
                <p class="login-logo">Praise</p>
                <v-divider dark></v-divider>
                <div class="lower-side">
                  <p class="first-lower">Alfi</p>
                  <p data-testid="second-lower" class="second-lower">
                    Lorem app
                  </p>
                </div>
              </div>
            </v-col>
          </v-row>
        </div>
      </v-col>

      <v-col xs="12" md="7">
        <div class="fill-height d-flex align-center">
          <v-row no-gutters style="max-width: 700px; margin: 0 auto">
            <v-col col="12">
              <div class="login-form-wraper">
                <v-form
                  ref="login_form"
                  :lazy-validation="true"
                  @submit.prevent="handleLogin"
                >
                  <p class="text-center" style="font-size: 12px; color: red">
                    {{ message }}
                  </p>
                  <v-text-field
                    rounded
                    outlined
                    dense
                    v-model="user.email"
                    :rules="rules.email"
                    :append-icon="'mdi-email'"
                    label="Email"
                    required
                  ></v-text-field>
                  <v-text-field
                    v-model="user.password"
                    :append-icon="password_show ? 'mdi-eye' : 'mdi-eye-off'"
                    :rules="rules.password"
                    :type="password_show ? 'text' : 'password'"
                    name="input-10-1"
                    label="Password"
                    hint="At least 4 characters"
                    rounded
                    outlined
                    dense
                    @click:append="password_show = !password_show"
                  ></v-text-field>

                  <v-checkbox
                    dense
                    style="margin-top: 0"
                    v-model="remember_me"
                    label="Remember me"
                  ></v-checkbox>
                  <v-row>
                    <v-col col="5">
                      <v-btn
                        data-testid="login-button"
                        :disabled="loading"
                        @click="handleLogin"
                        rounded
                        color="primary"
                        class="mr-4 pr-10 pl-10"
                      >
                        Login
                        <v-icon right dark small>mdi-arrow-right</v-icon>
                      </v-btn>
                    </v-col>
                    <v-col col="7" v-if="loading">
                      <v-progress-circular
                        :value="100"
                        :width="3"
                        color="blue-grey"
                        indeterminate
                      ></v-progress-circular>
                    </v-col>
                  </v-row>
                </v-form>
              </div>
            </v-col>
          </v-row>
        </div>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import { mapActions } from "vuex";
import User from "../models/user";

export default {
  name: "Login",
  data() {
    return {
      user: new User("", "", "", ""),
      loading: false,
      password_show: false,
      remember_me: false,
      message: "",
      rules: {
        email: [
          (v) => !!v || "Email is required",
          (v) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(v) || "E-mail must be valid",
        ],
        password: [(v) => (v && v.length >= 4) || "Min 4 characters"],
      },
    };
  },
  methods: {
    ...mapActions({ login: "auth/login" }),

    handleLogin() {
      this.loading = true;
      if (!this.$refs.login_form.validate()) {
        this.loading = false;
        return;
      }
      if (this.user.email && this.user.password) {
        let formData = new FormData();
        let rawData = JSON.stringify(this.user);
        formData.append("data", rawData);
        this.login(formData).then(
          () => {
            this.$router.push({ name: "settings_profile" });
          },
          () => {
            this.loading = false;
            this.message = "Account not found";
          }
        );
      }
    },
  },
};
</script>

<style lang="scss" scoped>
.bg-img-wrapper {
  background-image: url("~@/assets/images/tom-barrett.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  align-items: center;

  .top-logo-display {
    text-align: right;
    max-width: 400px;
    margin: 0 auto;
    .login-logo {
      line-height: 1;
      font-size: 40px;
      font-weight: 500;
      color: white;
      padding-right: 21px;
    }

    .lower-side {
      .first-lower {
        margin-top: 20px;
        margin-bottom: 0;
        line-height: 1;
        color: white;
        font-size: 20px;
        font-style: italic;
        font-weight: 100;
      }
      .second-lower {
        margin-bottom: 0;
        line-height: 1;
        color: white;
        font-size: 30px;
        font-weight: 500;
      }
    }
  }
}

.login-form-wraper {
  max-width: 350px;
}
</style>