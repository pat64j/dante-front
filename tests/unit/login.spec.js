import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Login from '@/views/Login.vue';


describe('Login.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(Vuetify)
    wrapper = shallowMount(Login, {
      localVue,
    });
  });

  it('should render the title and the login button', () => {
    // to make sure we are free from vuetify components in our tests, use data-testid attributes to search for elements
    // expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('[data-testid="second-lower"]').exists()).toBe(true);
    expect(wrapper.find('[data-testid="login-button"]').exists()).toBe(true);
  });

});