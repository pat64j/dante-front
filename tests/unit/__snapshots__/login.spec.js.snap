// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue should render the title and the login button 1`] = `
<v-container-stub tag="div" fluid="true" class="fill-height pa-0">
  <v-row-stub tag="div" nogutters="true" class="fill-height">
    <v-col-stub md="5" tag="div" class="d-none d-md-block">
      <div class="bg-img-wrapper fill-height">
        <v-row-stub tag="div" nogutters="true">
          <v-col-stub tag="div" col="12">
            <div class="top-logo-display"><img src="" contain="" height="150" width="150" class="my-3" style="margin: 0px;">
              <p class="login-logo">Praise</p>
              <v-divider-stub dark="true"></v-divider-stub>
              <div class="lower-side">
                <p class="first-lower">Alfi</p>
                <p data-testid="second-lower" class="second-lower">Church app</p>
              </div>
            </div>
          </v-col-stub>
        </v-row-stub>
      </div>
    </v-col-stub>
    <v-col-stub md="7" tag="div" xs="12">
      <div class="fill-height d-flex align-center">
        <v-row-stub tag="div" nogutters="true">
          <v-col-stub tag="div" col="12">
            <div class="login-form-wraper">
              <v-form-stub lazyvalidation="true">
                <p class="text-center" style="font-size: 12px; color: red;"></p>
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="v => !!v || 'Email is required',v => /^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$/.test(v) || 'E-mail must be valid'" successmessages="" appendicon="mdi-email" backgroundcolor="" dense="true" label="Email" loaderheight="2" clearicon="$clear" outlined="true" rounded="true" type="text" required=""></v-text-field-stub>
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="v => v &amp;&amp; v.length >= 4 || 'Min 4 characters'" successmessages="" appendicon="mdi-eye-off" backgroundcolor="" dense="true" hint="At least 4 characters" label="Password" loaderheight="2" clearicon="$clear" outlined="true" rounded="true" type="password" name="input-10-1"></v-text-field-stub>
                <v-checkbox-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" dense="true" label="Remember me" ripple="true" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" indeterminateicon="$checkboxIndeterminate" officon="$checkboxOff" onicon="$checkboxOn" style="margin-top: 0px;"></v-checkbox-stub>
                <v-row-stub tag="div">
                  <v-col-stub tag="div" col="5">
                    <v-btn-stub color="primary" rounded="true" tag="button" activeclass="" type="button" data-testid="login-button" class="mr-4 pr-10 pl-10">
                      Login
                      <v-icon-stub right="" dark="" small="">mdi-arrow-right</v-icon-stub>
                    </v-btn-stub>
                  </v-col-stub>
                  <!---->
                </v-row-stub>
              </v-form-stub>
            </div>
          </v-col-stub>
        </v-row-stub>
      </div>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
